### ISSUE TITLE
TRACKER - 16-bit depth viewports needed for improved flowmap calculation (BLOCKED by Godot 4.0)

### ISSUE THREAD

**Issue by Arnklit** [Arnklit/WaterGenGodot#19]
Godot 4.0 does not have  16-bit viewports yet, which means I cannot calculate flowmaps and various other textures correctly.

There is a PR working on it, but it is stale. https://github.com/godotengine/godot/pull/61667

EDIT: It turned out I had some bugs in my code that made the problem worse, fixing those issues made it doable to generate the flowmaps, thought they will be better with 16 bit viewports. I might also move away from using Viewports and use Render Device directly, but that will require quite a bit of work. 

Keeping the bug open now to track the issue.

**Comment by TokisanGames** [comment]
This proposal discusses plans for 16/32-bit viewports. 

https://github.com/godotengine/godot-proposals/issues/2935

11 days ago [here](https://github.com/godotengine/godot-proposals/issues/2935#issuecomment-1319314253) clayjohn said, "3D rendering still happens in HDR (16bpc)... We have a plan to expose a way to get the HDR buffer directly from the viewport... Ultimately though, for custom effects we expose the ability to create your own framebuffers and textures through the RenderingDevice directly so you don't have to go through the Godot pipeline."

Can you create your own frame buffer as he says?

**Comment by Arnklit** [comment]
Yes I think I can. Me and Rod (creator of Material Maker) have been discussing this. It will probably be needed for some of the more advanced stuff I want to add to Waterways, but for now I've actually gotten it to work with the 8-bit viewports.

**Comment by TokisanGames** [comment]
HDR viewports are now being used in the godot4 branch, aren't they?
